src/esp/scene/SceneNode.cpp

```diff
@@ -47,5 +53,29 @@ const Mn::Range3D& SceneNode::computeCumulativeBB() {
   return cumulativeBB_;
 }
 
+void SceneNode::clean(const Magnum::Matrix4& absoluteTransformation) {
+  worldCumulativeBB_ = Cr::Containers::NullOpt;
+
+  absoluteTransformation_ = absoluteTransformation;
+}
+
+Mn::Vector3 SceneNode::absoluteTranslation() const {
+  if (isDirty())
+    return absoluteTransformation().translation();
+  else
+    return absoluteTransformation_.translation();
+}
+
+const Mn::Range3D& SceneNode::getAbsoluteAABB() const {
+  if (aabb_)
+    return *aabb_;
+  else {
+    if (!worldCumulativeBB_)
+      worldCumulativeBB_ = {
+          geo::getTransformedBB(getCumulativeBB(), absoluteTransformation_)};
```

Comment:
Glad to see `geo::getTransformedBB` becomes useful in this performance-critical project. I remembered I optimized this function so that it became 13x faster.

Reply:
Yeah!  The existence of that function and it being very fast is what makes this work well :)

Reply:
> Logically, a class should not be a node and an attachment to that node at the same time. 

That's supported *by design*, and perfectly fine. (Unless your coding guidelines ban multiple inheritance.)

Could be also a member variable, but then everything becomes more verbose, and you'll need another derived class and...